### PR TITLE
Qbllr/mocha tester

### PR DESCRIPTION
**Pull request by qballer**
## Proposed Changes

  - Added mocha tester. 
There is a bug in bit which doesn't pass `context.componentDir` to `tester.getDynamicPackageDependencies` - issue solve https://github.com/teambit/bit/issues/1154
-  Added jest tester 

This PR currently breaks on CI with node version, I'm trying to figure out why:
npm 3 will not bring dev version unless hat is removed! 13.0.4-dev.4 and not ^13.0.4-dev.4. this will bring 13.0.4
finally: V8 5.8+ (e.g. in Node.js 8.1.4+): available behind the flag --harmony-promise-finally (details).

### CODE REVIEW

File: src/jest.ts [hunk 1]
```diff
@@ -24,31 +26,36 @@ export function CreateJestTester(): TesterExtension {
...
+                outputFile
             },[directory])
-                .then(function (results:any){
-                    // console.log('My INFO')
-                    // console.log('My INFO')
-                    // console.log('My INFO')
-                    // console.log(JSON.stringify(results.results.testResults))
-                    return results.results.testResults.map(convertJestFormatToBitFormat)
+                .then(function (){
+                    let results = JSON.parse(fs.readFileSync(outputFile).toString())
```
Comment (davidfirst) [inline src/jest.ts]: you might wanna use fs.readJson()

File: src/mocha/mocha.ts [hunk 2]
```diff
@@ -27,14 +27,21 @@ export function CreateMochaTester(): TesterExtension {
...
                         mocha.addFile(testFile.path)
                     })
                     mocha.run()
-                        .on('end', function (this: { testResults: any }) {
-                            const result = normalizeResults(this.testResults)
-                            return resolve([result])
-                        })
+                    .on('end', function (this: { testResults: any }) {
+                        const results = this.testResults
+                        const rawResults = ([]).concat.apply([], Object.keys(results).map(function (item) {
```
Comment (davidfirst) [inline src/mocha/mocha.ts]: why is this `([]).concat.apply([],` needed?

Comment (qballer) [reply src/mocha/mocha.ts]: to flatten the array of the result is was an ugly fast fix.

File: src/jest.ts [hunk 3]
```diff
@@ -3,6 +3,8 @@ import {loadPackageJsonSync, findByName, fillDependencyVersion} from './compiler
 import _get from 'lodash.get'
 import {runCLI} from 'jest-cli'
 import {convertJestFormatToBitFormat} from './result-adapter'
+import path from 'path'
+import fs from 'fs-extra'
```
[the remark this answers is not in the document]
Comment (qballer) [reply src/jest.ts]: Will be fixed in next PR

File: src/jest.ts (hunk 1 above)
Comment (qballer) [reply src/jest.ts]: I need a sync function. But I didn't know this one, awesome. (I can async this, but not for now)

Comment (davidfirst) [reply src/jest.ts]: Yeah, fs-extra has a sync method for this as well. fs.readJsonSync

Comment (qballer) [reply src/jest.ts]: Will be fixed next PR.

**Pull request merged**